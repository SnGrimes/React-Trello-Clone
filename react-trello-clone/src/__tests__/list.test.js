import React from 'react';
import { shallow} from 'enzyme'
import List from '../components/list';

describe('List', () => {
  const listTitle = 'Test1';
  const props = {key: listTitle, listTitle: listTitle};
  const list = shallow(<List {...props}/>);
  const mockAddCard = jest.fn();
  const mockShowForm = jest.fn();
  jest.mock('../components/list', () => {
    return jest.fn().mockImplementation(() => {
      return {addCard : mockAddCard, showForm: mockShowForm};
    });
  });
  
 
  it('renders correctly', () => {
    expect(list).toMatchSnapshot();
  });
  it('receives the props', () => {
    expect(list.instance().props.listTitle).toEqual(listTitle);
  });
  it('initializes `state` with an empty card array', () => {
    expect(list.state().cards).toEqual([]);
  });
  it('initializes `state` with newCard to false', () => {
    expect(list.state().newCard).toEqual(false);
  });
  describe('when clicking the `newCard`', () => {
    const cardDesc = 'This is a card.';
    beforeEach(() => {
      list.find('.list__new-card').simulate('click');      
    });
    it('calls `showForm` function on click', () => {
      expect(list.state().newCard).toEqual(true);
    });
    
  }); // click new card describe block

}); // List component describe block