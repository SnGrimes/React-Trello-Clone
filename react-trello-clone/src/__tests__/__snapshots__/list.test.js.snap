// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    listTitle="Test1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Test1
</h5>,
        <div>
          <button
                    className="list__new-card button small"
                    onClick={[Function]}
          >
                    Add a Card...
          </button>
</div>,
      ],
      "className": "cell small-3 small-offset-0 list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test1",
        },
        "ref": null,
        "rendered": "Test1",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="list__new-card button small"
            onClick={[Function]}
>
            Add a Card...
</button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add a Card...",
            "className": "list__new-card button small",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add a Card...",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Test1
</h5>,
          <div>
            <button
                        className="list__new-card button small"
                        onClick={[Function]}
            >
                        Add a Card...
            </button>
</div>,
        ],
        "className": "cell small-3 small-offset-0 list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test1",
          },
          "ref": null,
          "rendered": "Test1",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="list__new-card button small"
              onClick={[Function]}
>
              Add a Card...
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add a Card...",
              "className": "list__new-card button small",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add a Card...",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
